clang/lib/3C/ConstraintVariables.cpp

```diff
@@ -517,8 +520,9 @@ PointerVariableConstraint::PointerVariableConstraint(
   // Get a string representing the type without pointer and array indirection.
   BaseType = extractBaseType(D, TSInfo, QT, Ty, C);
 
-  IsVoidPtr = isTypeHasVoid(QT);
-  bool IsWild = !getIsGeneric() && (isVarArgType(BaseType) || IsVoidPtr);
+  IsVoidPtr = isPtrOrArrayType(QT) && isTypeHasVoid(QT);
+  bool IsWild = isVarArgType(BaseType) ||
+      (!(PotentialGeneric || getIsGeneric()) && IsVoidPtr);
```

Comment:
Would be worth adding a larger comment explaining the logic here.

Reply:
Yes, I struggled with this too much before tests. Do you know if a `VarArgType` can be generic? I assumed not. (As in, not explicitly annotated as anything more than it is.)

Reply:
I think we should assume varargs cannot be generic.